packages/nodes-base/nodes/RespondToWebhook/utils.ts

```diff
@@ -18,3 +22,47 @@ export const configuredOutputs = (
 
 	return ['main'];
 };
+
+export const replaceSingleQuotes = (responseBody: string) => {
+	return responseBody.replace(/'/g, '&#39;');
+};
+
+/*
+ * To prevent XSS and CSRF, we sanitize the response body by wrapping it in an iframe.
+ * This prevents the API from being accessed since the iframe origin will be empty.
+ */
+export const sanitizeResponseData = (responseBody: string, shouldReplaceSingleQuotes: boolean) => {
+	const parsedResponseBody = shouldReplaceSingleQuotes
+		? replaceSingleQuotes(responseBody)
+		: responseBody;
+
+	return `
+		<iframe srcdoc='${parsedResponseBody}'
+			style="position:fixed; top:0; left:0; width:100vw; height:100vh; border:none; overflow:auto;"
+			allowtransparency="true">
+		</iframe>`;
+};
+
+export const getBinaryResponse = (binaryData: IBinaryData, headers: IDataObject) => {
+	let responseBody: IN8nHttpResponse | Readable;
+
+	if (binaryData.id) {
+		responseBody =
+			binaryData.mimeType === 'text/html'
```

Comment:
We need to include also at least `application/xhtml+xml` as Firefox treats that as HTML.